.gitlab/ci/on-push.yml

```diff
@@ -474,4 +474,40 @@ xsiam_server_ga:
   variables:
     INSTANCE_ROLE: "XSIAM 1.2"
     GCS_QUEUE_FILE: "queue-ga"
-    TEST_MACHINES_LIST: "test-machines-ga"
\ No newline at end of file
+    TEST_MACHINES_LIST: "test-machines-ga"
+
+test_upload_flow:
+  tags:
+    - gke
+  extends:
+    - .default-job-settings
+  rules:
+    - if: '$CI_PIPELINE_SOURCE =~ /^(push|contrib)$/'
+  variables:
+    CURRENT_BRANCH: "$CI_COMMIT_BRANCH"
+  stage: unittests-and-validations
+  script:
+    - |
+      if !$(git diff origin/master upload_testing_flow --name-only | grep -q "Tests/\|Utils/") ; then                                                                            16:33:03
```

Comment:
don't forget to replace `upload_testing_flow` with the current branch name

Reply:
done